### PR TITLE
agent: do not follow link when mounting container proc and sysfs

### PR DESCRIPTION
**Pull request by bergwolf**
Attackers might use it to explore other containers in the same pod.
While it is still safe to allow it, we can just close the race window
like runc does.

Fixes: #885

ref: https://github.com/opencontainers/runc/pull/2207

### CODE REVIEW

**Review by liubin — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
lgtm!

File: src/agent/rustjail/src/mount.rs [hunk 1]
```diff
@@ -205,6 +205,20 @@ pub fn init_rootfs(
...
 
+            // If the destination already exists and is not a directory, we bail
+            // out This is to avoid mounting through a symlink or similar -- which
+            // has been a "fun" attack scenario in the past.
+            if m.r#type == "proc" || m.r#type == "sysfs" {
+                if let Ok(meta) = fs::symlink_metadata(&m.destination) {
+                    if !meta.is_dir() {
+                        return Err(anyhow!(
+                            "Mount point {} must be ordinary directory",
+                            m.destination
```
Comment (jodh-intel) [inline src/agent/rustjail/src/mount.rs:216]: How about:

```suggestion
                        return Err(anyhow!(
                            "Mount point {} must be ordinary directory, got: {:?}",
                            m.destination,
                            meta.file_type()
```

The value displayed is in decimal but still useful for bug reports as we can simply convert to octal ;)

Comment (bergwolf) [reply src/agent/rustjail/src/mount.rs]: fixed, thanks!

**Review by jodh-intel — APPROVED** [review]
Thanks @bergwolf.

lgtm

**Pull request merged**